src/Pages/UserAccount/DeliveryPage.js

```diff
@@ -0,0 +1,322 @@
+import { useQuery } from "react-query";
+import { auth } from "../../FirebaseConfigs/Firesbase";
+import { DeliveryAddressSchema } from "../../ValidationSchemas/DeliverySchema";
+import DeliveryCardItem from "./Components/DeliveryCardItem";
+import UseAnimation from "../../Components/Loader";
+import loading from "react-useanimations/lib/loading";
+import { database } from "../../FirebaseConfigs/Firesbase";
+import { ref, set, onValue, push, remove } from "firebase/database";
+import {
+  DeliveryAddressService,
+  DeliveryServices,
+} from "../../Services/AccountServices";
+
+// TODO: FIX EDIT AND DELETE DELIVERY HANDLERS
+
+export default function DeliveryPage({ deliveryId }) {
+  const [modal, setModal] = useState(false);
+  const [editModal, setEditModal] = useState(false);
+  const [user, setUser] = useState(null);
+  const [userId, setUserId] = useState(null);
+  const [firstName, setFirstName] = useState("");
+  const [lastName, setLastName] = useState("");
+  const {
+    data = [],
+    isLoading,
+    error,
+    isError,
+    refetch,
+  } = useQuery(["delivery", userId], () => DeliveryServices(userId));
+  console.log(data);
+  console.log("userId:", userId);
+
+  const { data: address } = useQuery(["address", userId, deliveryId], () =>
+    DeliveryAddressService(userId, deliveryId)
+  );
+  console.log(`Alone data: ${address}`);
+
+  useEffect(() => {
+    const unsubscribe = onAuthStateChanged(auth, (data) => {
+      if (data) {
+        setUser(data.displayName);
+        setUserId(data.uid);
+        const [first, last] = data.displayName.split(" ");
+        setFirstName(first);
+        setLastName(last);
+      } else {
+        setUser(null);
+      }
+    });
+    return () => {
+      unsubscribe();
+    };
+  }, []);
+
+  const submitAddressHandler = (values, actions) => {
+    const db = database;
+    const newDeliveryRef = push(ref(db, userId + "/delivery/"));
+    set(newDeliveryRef, {
+      firstName: values.firstName,
+      lastName: values.lastName,
+      address: values.address,
+      apt: values.aptSuite,
+      zip: values.zip,
+      city: values.city,
+      state: values.state,
+    })
+      .then(() => {
+        const deliveryId = newDeliveryRef.key;
+
+        alert(deliveryId);
+        actions.resetForm({
+          values: {
+            firstName: "",
+            lastName: "",
+            address: "",
+            aptSuite: "",
+            zip: "",
+            city: "",
+            state: "",
+          },
+        });
+        setModal(false);
+      })
+      .catch((error) => {
+        alert(error);
+      });
+  };
+
+  const deleteDeliveryHandler = async (userId, deliveryId) => {
+    const dbRef = ref(database, userId + "/delivery/" + deliveryId);
+    remove(dbRef)
+      .then(() => {
+        alert("Deleted successfully");
+        refetch();
+      })
+      .then((error) => {
+        alert(`delivery error ${error}`);
+      });
+  };
+
+  const editHandler = () => {
```

Comment:
Fix editHandler method...
1- To retrieve data allocated to a single delivery address